File: sender/sender.js
var http=require("http"),
    fs = require("fs");

var postData = fs.readFileSync("postData.xml")


var req = http.request({
    hostname:"localhost",
    port: 9000,
    method:"POST",
    path: "/grounding/12345",
    headers: {
        "Content-Type" : "application/soap+xml; charset=utf-8",
        "Content-Length": postData.length
    },
},function(res){
    console.log('STATUS: ' + res.statusCode);
    console.log('HEADERS: ' + JSON.stringify(res.headers));
    res.setEncoding('utf-8');
    res.on('data',function(chunk){
        console.log("BODY: " + chunk);
    });
});

req.on("error", function(err){
    console.log("Problem with request: " + err.message);
});

req.write(postData);
req.end();
console.log("message sent");

File: sender/sender.go
package main

import (
    "bytes"
    "fmt"
    "net/http"
    "io/ioutil"
)

func main() {
    url := "http://localhost:9001/grounding/12345"
    fmt.Println("URL:>", url)

    var postData = []byte(`<?xml version="1.0" encoding="utf-8"?><root><element></element></root>`)
    req, err := http.NewRequest("POST", url, bytes.NewBuffer(postData))
    req.Header.Set("Content-Type", "application/soap+xml; charset=utf-8")
    req.Header.Set("Content-Length", len(postData))

    client := &http.Client{}
    resp, err := client.Do(req)
    if err != nil {
        panic(err)
    }
    defer resp.Body.Close()

    fmt.Println("response Status:", resp.Status)
    fmt.Println("response Headers:", resp.Header)
    body, _ := ioutil.ReadAll(resp.Body)
    fmt.Println("response Body:", string(body))
}
    